### PR TITLE
draft of an enhanced LearningRateScheduler

### PR DESCRIPTION
**Pull request by marcuniq**
I enhanced the LearningRateScheduler callback such that different learning rate schedules can be implemented.

You can still provide a simple function, which takes the epoch or batch as parameter, or you can provide an instance of the new class LearningRateSchedule.

What do you think?

### CODE REVIEW

**Comment by EderSantana** [comment]
I think you can make your `scheduler` as a class with a `__call__` method and implement all your suggestions with the current callback. To maintain a new class for each and every schedule won't scale well, I think. 

Let me know if I missed anything.

File: keras/utils/generic_utils.py [hunk 1]
```diff
@@ -68,49 +69,61 @@ def update(self, current, values=[]):
...
+            # print eta
+            if 'eta' in self.output_values:
+                if self.seen_so_far < self.target:
+                    info += ' - ETA: %ds' % self.output_values['eta']
                 else:
-                    info += ' - %s: %s' % (k, self.sum_values[k])
+                    info += ' - %ds' % (now - self.start)
+
+            # print all other values
+            for k, v in filter(lambda (x, _): x != 'prog' and x != 'eta', self.output_values.items()):
```
Comment (fchollet) [inline keras/utils/generic_utils.py]: Invalid Python 3 syntax.

**Comment by fchollet** [comment]
Checks appear to be failing: no Python 3 compatibility. 

> I think you can make your scheduler as a class with a __call__ method and implement all your suggestions with the current callback. To maintain a new class for each and every schedule won't scale well, I think.

Yes, that sounds reasonable.

**Comment by franciscovargas** [comment]
does the CLR sheduler work ???  with the latest build of keras ?? (the triangular learning sheduler)
